core/services/log/orm.go

```diff
@@ -81,17 +129,118 @@ func (o *orm) MarkBroadcastConsumed(blockHash common.Hash, blockNumber uint64, l
 		return errors.Wrap(err, "MarkBroadcastConsumed failed")
 	}
 	if rowsAffected == 0 {
-		return errors.Errorf("cannot mark log broadcast as consumed: does not exist")
+		return errors.Wrap(err, "while marking log broadcast as consumed")
+	}
+	return nil
+}
+
+func (o *orm) RemoveUnconsumedSetPending(qopts ...postgres.QOpt) (*int64, error) {
+	minUnconsumed, err := o.getUnconsumedMinBlock(qopts...)
+	if err != nil {
+		return nil, err
+	}
+	minPending, err := o.GetBroadcastsPending(qopts...)
+	if err != nil {
+		return nil, err
+	}
+	if minUnconsumed == nil {
+		return minPending, nil
+	}
+	if minPending == nil || *minUnconsumed < *minPending {
+		minPending = minUnconsumed
+		if err := o.SetBroadcastsPending(minPending, qopts...); err != nil {
+			return nil, err
+		}
+	}
+	if err := o.removeUnconsumed(qopts...); err != nil {
```

Comment:
Remove the unconsumed entries only after `SetBroadcastsPending` so that we don't lose information if we crash from this method.

Reply:
This whole set of ops was originally one gnarly atomic SQL query, but it was slow and not strictly necessary since we can just perform them in this idempotent ordering instead.

Reply:
I think this function could use a few well place comments

Reply:
Would it benefit from wrapping in a transaction or nah?

Reply:
It's not necessary with this ordering, but I will document it better :+1: 
The only benefit that I saw was some potential optimization from combining the select and the delete by returning block_number, but that might need extra setup like a view to index properly.  What might a worst case scenario look like? Hundreds of unconsumed rows from jobs in flight during a crash?